astropy/coordinates/angles.py

```diff
@@ -553,11 +553,17 @@ def _validate_angles(self, angles=None):
         if angles.unit is u.deg:
             limit = 90
         elif angles.unit is u.rad:
-            limit = self.dtype.type(0.5 * np.pi)
+            limit = 0.5 * np.pi
         else:
             limit = u.degree.to(angles.unit, 90.0)
 
-        invalid_angles = np.any(angles.value < -limit) or np.any(angles.value > limit)
+        # Ensure ndim>=1 so that comparison is done using the angle dtype.
+        # Otherwise, e.g., np.array(np.pi/2, 'f4') > np.pi/2 will yield True.
+        # (This feels like a bug -- see https://github.com/numpy/numpy/issues/23247)
```

Comment:
It looks like since this was written there might be a fix in a future numpy for this - might want to add a todo note aobut that

Reply:
xref https://github.com/numpy/numpy/issues/23247